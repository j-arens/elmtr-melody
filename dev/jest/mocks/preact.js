module.exports = {
    unmountComponent: jest.fn(),
};
