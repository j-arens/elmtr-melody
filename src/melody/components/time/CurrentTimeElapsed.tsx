import { WithOptionalClassName } from '@components/type';
import { cySelector } from '@utils/index';
import { h } from 'preact';
import TimeProvider from './TimeProvider/';

interface Props extends WithOptionalClassName {
    currentTime: number;
    getTime: (time: number) => string;
}

export const CurrentTimeElapsed = ({ currentTime, getTime, className = '' }: Props) => (
    <span class={className} {...cySelector('time-elapsed')}>
        {getTime(currentTime)}
    </span>
);

export default TimeProvider(CurrentTimeElapsed);
