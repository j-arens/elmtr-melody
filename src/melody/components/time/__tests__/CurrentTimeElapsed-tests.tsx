import { h } from 'preact';
import { render } from 'preact-render-to-string';
import { CurrentTimeElapsed } from '../CurrentTimeElapsed';

describe('CurrentTimeElapsed', () => {
    it('renders', () => {
        const component = render(
            <CurrentTimeElapsed
                className="alpine-white"
                currentTime={0}
                getTime={jest.fn().mockReturnValueOnce(80)}
            />,
        );
        expect(component).toMatchSnapshot();
    });
});
