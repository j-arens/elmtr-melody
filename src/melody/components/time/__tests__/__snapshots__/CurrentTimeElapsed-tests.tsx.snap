// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrentTimeElapsed renders 1`] = `<span class="alpine-white" data-cy="time-elapsed">80</span>`;
